sphinx/domains/cpp.py

```diff
@@ -543,6 +545,18 @@
 }
 
 
+class _Placement(enum.Enum):
+    Left = 'left'
+    Right = 'right'
+
+
+def _check_placement_arg(val: str) -> _Placement:
+    if val not in ('left', 'right'):
+        raise RuntimeError('Invalid C++ placement parmeter value '
+                           '"{}", should be either "left" or "right"'.format(val))
+    return _Placement(val)
```

Comment:
Marginally faster as
```suggestion
    try:
        return _Placement(val)
    except ValueError:
        raise RuntimeError('Invalid C++ placement parmeter value '
                           '"{}", should be either "left" or "right"'.format(val)) from None
```